projects/cds/src/merchandising/converters/merchandising-products-normalizer.spec.ts

```diff
@@ -0,0 +1,148 @@
+import { ConverterService, ImageType } from '@spartacus/core';
+import { MerchandisingProducts } from '../model/merchandising.products';
+import { StrategyResult } from '../model/strategy.result';
+import { MerchandisingProductsNormalizer } from './merchandising-products-normalizer';
+import createSpy = jasmine.createSpy;
+
+const STRATEGY_RESULT_METADATA: Map<string, string> = new Map<string, string>();
+STRATEGY_RESULT_METADATA.set('test-metadata-field', 'test-metadata-value');
+const PRODUCT_METADATA: Map<string, string> = new Map<string, string>();
+PRODUCT_METADATA.set(
+  'test-product-metadata-field',
+  'test-product-metadata-field'
+);
+const STRATEGY_RESULT: StrategyResult = {
+  resultCount: 1,
+  products: [
+    {
+      id: 'test-product-id',
+      name: 'test-product',
+      description: 'test-product',
+      brand: 'test-brand',
+      pageUrl: 'http://some-product-url',
+      thumbNailImage: 'http://some-thumbnail-imgae-url',
+      mainImage: 'http://some-main-imgae-url',
+      price: 20.99,
+      metadata: PRODUCT_METADATA,
+    },
+  ],
+  paged: {
+    from: 1,
+    size: 5,
+  },
+  metadata: STRATEGY_RESULT_METADATA,
+};
+
+const MERCHANDISING_PRODUCTS: MerchandisingProducts = {
+  metadata: STRATEGY_RESULT_METADATA,
+  products: [
+    {
+      code: 'test-product-id',
+      name: 'test-product',
+      price: {
+        formattedValue: '20.99',
+        value: 20.99,
+      },
+      images: {
+        PRIMARY: {
+          product: {
+            url: 'http://some-main-imgae-url',
+            format: 'product',
+            imageType: ImageType.PRIMARY,
+          },
+        },
+      },
+    },
+  ],
+};
+
+class MockConverterService {
+  convert = createSpy().and.returnValue(MERCHANDISING_PRODUCTS.products[0]);
+}
+
+describe('MerchandisingProductsNormalizer', () => {
+  let productsNormalizer: MerchandisingProductsNormalizer;
+  let converterService: ConverterService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        {
+          provide: ConverterService,
+          useClass: MockConverterService,
+        },
+        MerchandisingProductsNormalizer,
+      ],
+    });
+
+    converterService = TestBed.get(ConverterService as Type<ConverterService>);
+    productsNormalizer = TestBed.get(MerchandisingProductsNormalizer as Type<
+      MerchandisingProductsNormalizer
+    >);
+  });
+
+  it('should be created', () => {
+    expect(productsNormalizer).toBeTruthy();
+  });
+
+  it('should convert a StrategyResult with no metadata to a MerchandisingProducts with no metadata', () => {
+    const NO_METADATA_STRATEGY_RESULT = {
+      ...STRATEGY_RESULT,
+      metadata: null,
+    };
+
+    const NO_METADATA_MERCHANDISING_PRODUCTS = {
+      ...MERCHANDISING_PRODUCTS,
+    };
+    // metadata should not just be null/undefined on the converted MerchandisingProducts, it should not exist at all
+    delete NO_METADATA_MERCHANDISING_PRODUCTS.metadata;
```

Comment:
Hmm, not sure about this approach. How about having `MERCHANDISING_PRODUCTS` without `metadata`, and then create another variable and add this property to it?

Reply:
I've corrected this approach